### PR TITLE
Fix low color contrast issues across several widgets

### PR DESCRIPTION
**Pull request by Myranae**
## Summary:
Worked on as part of GAAD. 
The issue here was specific to the low contrast found for the dot and line in number-line. This fix changes the KhanColors object in Perseus which is used across a number of widgets. The changes here specifically affect grapher, number-line, interaction, and interactive-graph (based on the updated snapshots).

[Discussion about what colors to use. Includes input from a designer](https://khanacademy.slack.com/archives/C01AZ9H8TTQ/p1715881111389849?thread_ts=1715880696.430059&cid=C01AZ9H8TTQ)

Before:
image
image
image

After:
image
image
image

Here's also some manual testing of how it looks in webapp (test everything course). Checked the other three types of graphs also affected by this as well.

Before:
image

After:
image

Issue: LEMS-252

## Test plan:
- Check all the widgets that were affected in Storybook and confirm the colors have at least 3:1 contrast with surroundings
- Also confirm the designs are not negatively affected
- Confirm all checks pass

### CODE REVIEW

File: packages/perseus/src/util/colors.ts
Comment (Myranae) [inline packages/perseus/src/util/colors.ts:87]: Could also just combine these into one since they are both the same color. Thoughts?

File: packages/perseus/src/util/colors.ts
Comment (Myranae) [inline packages/perseus/src/util/colors.ts:87]: Got rid of this part since both comments seem to think it is not needed. Replaced where they are used with just `"transparent"`.

**Comment by catandthemachines** [comment]
Do you have screen shots of the UX changes with the new colors? :)

**Comment by catandthemachines** [comment]
Were you able to run an accessibility scan and confirm the color doesn't have any more issues?

**Comment by catandthemachines** [comment]
Not sure how it works here just yet, but were you able to confirm or check with a designer on these new colors?

**Comment by Myranae** [comment]
> Do you have screen shots of the UX changes with the new colors? :)

Ooo, thank you! Will get these added :D

**Comment by Myranae** [comment]
> Were you able to run an accessibility scan and confirm the color doesn't have any more issues?

@catandthemachines Hmm, I'm not sure about an accessibility scan. How would I do that and what other issues are possible? I did triple check the ratio was better. I noticed the ratio between the lines and the dots might not be ideal, but not sure if that is fixable since the dot color has to have good contrast with the background (and so does the line).

**Comment by Myranae** [comment]
> Not sure how it works here just yet, but were you able to confirm or check with a designer on these new colors?

@catandthemachines I added a link to the PR summary that leads to the conversation I had with the team and a designer about what colors to use. Planning to make a ZND and check with our QE person on the change as well. (Before merging but after things slow down a bit).

**Comment by catandthemachines** [comment]
> > Were you able to run an accessibility scan and confirm the color doesn't have any more issues?
> 
> @catandthemachines Hmm, I'm not sure about an accessibility scan. How would I do that and what other issues are possible? I did triple check the ratio was better. I noticed the ratio between the lines and the dots might not be ideal, but not sure if that is fixable since the dot color has to have good contrast with the background (and so does the line).

There are a lot of great accessibility scan tools for Chrome browsers (I use Accessibility Insights when I want to confirm is my site has any issues or not https://chromewebstore.google.com/detail/accessibility-insights-fo/pbjjkligggfmakdaogkfomddhfmpjeni ). For the question on the dot vs. the line, that might be a great question for Level Access to give us suggestions. I can't remember off the top of my head but I believe distinguishing content in a visual diagram has different color contrast requirements than text vs. background.

**Review by catandthemachines — APPROVED** [review]

File: packages/perseus/src/util/colors.ts
Comment (Myranae) [reply packages/perseus/src/util/colors.ts:87]: Hmm, there are 45 instances of INTERACTIVE and 26 instances of INTERACTING all across 13 files. I would probably make `INTERACTING` just be `INTERACTIVE`, so it would only add 5 files with changes. 

I don't think it would have wider consequences since both are using the same color, though I don't know if the change in meaning would be troublesome. 

Perhaps this should be part of a wider change of updating `KhanColors` at some point and saved for another time since it has so many instances. I think I assumed it would be a smaller change. 😅

Comment (catandthemachines) [reply packages/perseus/src/util/colors.ts:87]: Yeah in that case I would hold off and just submit a backlog task to revisit in the future 😉

File: packages/perseus/src/widgets/number-line.tsx
Comment (nishasy) [inline packages/perseus/src/widgets/number-line.tsx:477]: How would you feel about replacing this (and other uses below) with WB `color.blue` and not importing `KhanColors` at all where possible? I think that would cut out the intermediate step and make it clear exactly where the color is coming from.

**Review by nishasy — APPROVED** [review]
Left a question/suggestion about using WB color instead of KhanColors in the files directly, but otherwise LGTM!

File: packages/perseus/src/widgets/number-line.tsx
Comment (Myranae) [reply packages/perseus/src/widgets/number-line.tsx:477]: I think I may have used some of my understanding of backend coding here to make this decision, but I think it might not be the same for frontend. Let me see what you think. I was thinking having one place where we import WB colors would be more efficient than importing it in all the places the colors are used, specifically because WB is an external library. So we're only doing the import here and not all over. Maybe the import still ends up happening in all the places anyway as each time it goes to to the KhanColors file and it ends up importing from WB at that point anyway. 🤔

**Pull request merged**
